src/lib/server/generateStaticHTMLFile.js

```diff
@@ -0,0 +1,33 @@
+import path from "path";
+import { prepareOutput } from "./RequestHandler";
+import { getLogger } from "./logger";
+import WebpackIsomorphicTools from "webpack-isomorphic-tools";
+import webpackConfig from "../../config/webpack-isomorphic-tools-config";
+import detectEnvironmentVariables from "../detectEnvironmentVariables";
+
+const LOGGER = getLogger();
+
+export default async function () {
+  const mockReq = {
+    headers: {
+      host: "localhost"
+    },
+    url: "/"
+  };
+
+  LOGGER.debug("about to run middleware");
+  global.webpackIsomorphicTools = new WebpackIsomorphicTools(webpackConfig)
+    .development(false)
+    .server(process.cwd(), async () => {
+      try {
+        const envVars = detectEnvironmentVariables(path.join(process.cwd(), "src", "config", "application.js"));
+        const Index = require(path.join(process.cwd(), "Index.js")).default;
+        const output = await prepareOutput(mockReq, {Index}, {}, {}, envVars, true);
+        LOGGER.debug(output.responseString);
```

Comment:
Currently I'm logging the responseString but really you want to write this to a file. The file would be `path.join(process.cwd(), "build", "index.html")` and you can do that easily with [fs.writeFileSync](https://nodejs.org/api/fs.html#fs_fs_writefilesync_file_data_options) we are okay using the synchronous version of `fs.writeFile` because this part is still under the command line interface. It is not used to process multiple incoming requests, so a blocking operation just makes the code easier to use without worrying about performance issues.